src/components/surveys/survey-design/question-edit/rhs-subcontrols/YearRadioGroup.tsx

```diff
@@ -0,0 +1,112 @@
+import {theme} from '@style/theme'
+import React from 'react'
+
+export const DEFAULT_MIN_YEAR = 1900
+export const DEFAULT_MAX_YEAR = new Date().getFullYear()
+
+export type YearFormatType = 'ANY' | 'CURRENT' | 'SET'
+
+const capitalizeFirstLetter = (val: string) => {
+  return `${val.charAt(0).toUpperCase()}${val.substring(1)}`
+}
+
+const yearFormatToAllowValueMap: Record<YearFormatType, boolean | undefined> = {
+  SET: undefined,
+  ANY: true,
+  CURRENT: false,
+}
+
+function yearFormatToAllowValue(yearFormat: YearFormatType) {
+  return yearFormatToAllowValueMap[yearFormat]
+}
+
+function allowValueToYearFormat(allowValue: boolean | undefined): YearFormatType {
+  return (
+    (Object.keys(yearFormatToAllowValueMap) as YearFormatType[]).find(
+      key => yearFormatToAllowValueMap[key] === allowValue
+    ) || 'ANY'
+  )
+}
+
+function setYearValue(yearFormat: YearFormatType, yearValue: number | undefined) {
+  return yearFormat === 'SET' ? yearValue : undefined
+}
+
+type YearRadioGroupProps = {
+  type: 'min' | 'max'
+  allowValue: boolean | undefined
+  yearValue: number | undefined
+  hasError: boolean
+  onChange: (allowValue: boolean | undefined, yearValue: number | undefined) => void
+}
+
+const YearRadioGroup: React.FunctionComponent<YearRadioGroupProps> = ({
+  type,
+  allowValue,
+  yearValue,
+  hasError,
+  onChange,
+}) => {
+  const id = `${type}YearGroup`
+  const CONFIG = {
+    min: {
+      tense: 'past',
+      yearLabel: 'Minimum Year',
+    },
+    max: {
+      tense: 'future',
+      yearLabel: 'Maximum Year',
+    },
+  }
+  const yearFormat: YearFormatType = allowValueToYearFormat(allowValue)
+  const defaultYearValue = type === 'min' ? DEFAULT_MIN_YEAR : DEFAULT_MAX_YEAR
+
+  const inlineInput = (
+    <Box display="flex" alignItems="center">
+      {`Set ${capitalizeFirstLetter(type)} Value`}
+      <SimpleTextInput
+        sx={{width: '80px', ml: '12px'}}
+        id={`${type}YearValue`}
+        error={hasError}
+        value={yearValue ?? ''}
+        type="number"
+        disabled={yearValue === undefined}
+        onChange={(e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+          onChange(
+            yearFormatToAllowValue(yearFormat),
+            setYearValue(yearFormat, parseInt(e.target.value) || defaultYearValue)
+          )
+        }}
+      />
+    </Box>
+  )
+
+  return (
+    <>
+      <FormControl fullWidth sx={{marginTop: 2}} error={hasError}>
+        <SimpleTextLabel id={id} sx={{marginBottom: 1}}>{`${CONFIG[type].yearLabel}`}</SimpleTextLabel>
+        <RadioGroup
+          aria-labelledby={id}
+          id={`${type}YearFormat`}
+          value={yearFormat}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+            const newYearFormat = e.target.value as YearFormatType
+            onChange(yearFormatToAllowValue(newYearFormat), setYearValue(newYearFormat, yearValue || defaultYearValue))
```

Comment:
Use `onChange` callback to send the updated allowValue and yearValue to the parent component.